### ISSUE TITLE
How to add to UserStorage with Kotlin

### ISSUE THREAD

**Issue by alisonthemonster** [actions-on-google/actions-on-google-java#17]
Is there a way to modify `UserStorage` when using Kotlin? 

`UserStorage` is a `Map` in [this implementation](https://github.com/actions-on-google/actions-on-google-java/blob/master/src/main/kotlin/com/google/actions/api/ActionRequest.kt#L93) and is not mutable. This doesn't seem to be an issue for Java projects (example [here](https://github.com/actions-on-google/dialogflow-name-psychic-java/blob/a69068c2d749836cd544188499560f0dc8ac1519/src/main/java/com/example/MyActionsApp.java#L136))

Thank you!

**Comment by taycaldwell** [comment]
Hi @alisonthemonster,

Thank you for the report. This seems to be an issue with the API returning an immutable Map as you mentioned.

I will look into a fix to update this for Kotlin developers using User Storage.

Thanks,

**Comment by zonder129** [comment]
Hi!
Is there still no solution for that issue? Maybe there is a workaround, cause I can't use userStorage for Kotlin. :(

**Comment by Fleker** [comment]
You can create a new map for the response and clone the data from the request to the response. This would probably be the better approach, as it would keep the Request immutable.

```
val userStorage = mutableMapOf<String, Any>()
userStorage.putAll(aogRequest.userStorage)
userStorage.set("key", "value")
val responseBuilder = ResponseBuilder(usesDialogflow = false, userStorage = userStorage)
val aogResponse = responseBuilder.build()
```

**Comment by zonder129** [comment]
@Fleker looks like there is another bug - aogRequest.userStorage is always empty, but aogRequest.user?.userStorage, which is String, is implemented. I am supposed to create a Map, parse a String, and only then I may use your approach. Can I add another issue for that?

**Comment by Fleker** [comment]
If your userStorage already exists, it should automatically [parse your string from JSON to a Map](https://github.com/actions-on-google/actions-on-google-java/blob/master/src/main/kotlin/com/google/actions/api/impl/AogRequest.kt#L308).

**Comment by zonder129** [comment]
Hi @Fleker, is there any update coming to solve this bug? There is still no way to properly clone userStorage map or create your own ResponseBuilder, as you did in the example above. Try it your self:
```kotlin
val actionRequest = MockRequestBuilder().setUserStorage(mapOf(Pair("userId","someId"))).build()
val userStorage = mutableMapOf<String, Any>()
userStorage.putAll(actionRequest.userStorage)
userStorage["key"] = "value"
println(userStorage)
```
The output will be only:
{key=value}

ResponseBuilder is also unavailable to be created because **its constructor is internal.** So there is no way to create it outside `com.google.actions.api.response` package.

**Comment by Fleker** [comment]
I was writing the code as a library test, so I must've missed the internal constructor issue.

**Comment by taycaldwell** [comment]
Hi all,

We identified that the workaround is not feasible and are working to address this issue. In the meantime, please open new issues for any other unrelated bugs.

Thanks!

**Comment by taycaldwell** [comment]
Hello,

This issue has been addressed as of [v1.7.0](https://github.com/actions-on-google/actions-on-google-java/releases/tag/v1.7.0).

If you run into any other issues, please open a new issue.

Thanks.
